client/src/components/Map/components/SuperBlock.js

```diff
@@ -79,21 +79,48 @@ export class SuperBlock extends Component {
                   .join('-') === blockDashedName
             )}
             key={blockDashedName}
+            noCurrentChallenge={noCurrentChallenge}
           />
         ))}
       </ul>
     );
   }
 
   render() {
-    const { superBlock, isExpanded, toggleSuperBlock } = this.props;
+    const {
+      superBlock,
+      isExpanded,
+      toggleSuperBlock,
+      nodes,
+      currentChallengeId
+    } = this.props;
+    const blocksForSuperBlock = nodes.filter(
+      node => node.superBlock === superBlock
+    );
+    const noCurrentChallenge = !nodes.some(
+      challenge => challenge.id === currentChallengeId
+    );
+    // If, somehow, the user does not have a current challenge, this opens the
+    // first superBlock.
+    const hasCurrentChallenge = noCurrentChallenge
+      ? superBlock === 'Responsive Web Design'
+      : blocksForSuperBlock.some(
+          challenge => challenge.id === currentChallengeId
+        );
+    const isOpen =
+      typeof isExpanded === 'undefined' ? hasCurrentChallenge : isExpanded;
     return (
-      <li className={`superblock ${isExpanded ? 'open' : ''}`}>
-        <div className='map-title' onClick={() => toggleSuperBlock(superBlock)}>
+      <li className={`superblock ${isOpen ? 'open' : ''}`}>
+        <div
+          className='map-title'
+          onClick={() => toggleSuperBlock(superBlock, !isOpen)}
```

Comment:
`toggleSuperBlock` didn't need to know the current expanded state before. I would be wary of this and ask why it needed to be changed.

Stateless is easier to maintain.